database/sql/src/main/scala/cromwell/database/slick/tables/MetadataEntryComponent.scala

```diff
@@ -109,17 +109,20 @@ trait MetadataEntryComponent {
     } yield metadataEntry).sortBy(_.metadataTimestamp)
   )
 
+  // This is only used for metadata summary which should not require metadata sorting if rows are committed
+  // with monotonically increasing IDs. The metadata summary logic records the maximum ID it last saw and uses
+  // that last ID + 1 as the minimum ID for the next query iteration.
   val metadataEntriesForIdGreaterThanOrEqual = Compiled(
     (metadataEntryId: Rep[Long], startMetadataKey: Rep[String], endMetadataKey: Rep[String], nameMetadataKey: Rep[String],
-     statusMetadataKey: Rep[String], likeLabelMetadataKey: Rep[String]) => (for {
+     statusMetadataKey: Rep[String], likeLabelMetadataKey: Rep[String]) => for {
       metadataEntry <- metadataEntries
       if metadataEntry.metadataEntryId >= metadataEntryId
       if (metadataEntry.metadataKey === startMetadataKey || metadataEntry.metadataKey === endMetadataKey ||
         metadataEntry.metadataKey === nameMetadataKey || metadataEntry.metadataKey === statusMetadataKey ||
         metadataEntry.metadataKey.like(likeLabelMetadataKey)) &&
         (metadataEntry.callFullyQualifiedName.isEmpty && metadataEntry.jobIndex.isEmpty &&
           metadataEntry.jobAttempt.isEmpty)
-    } yield metadataEntry).sortBy(_.metadataTimestamp)
+    } yield metadataEntry
```

Comment:
ToL: Looks like this was sorting by timestamp not insertion ID. Is this safe against situations where status updates get re-ordered on the wire?

Reply:
I believe we're robust to out of order events for everything, but definitely for status

Reply:
The type of events being summarized are not aggregated based on "most recent" so we don't need to sort them by timestamp.
But for the rest of metadata we definitely DO want them to be sorted by timestamp.

Reply:
We should be safe to reordering on the wire without sorting as long as we're not limiting the size of batches to process. And status is definitely safe thanks to the magic of CRDTs, while the other metadata only appears to be published from one spot for each.

Reply:
💯 - sorting a "journal"-style table does indeed seem like an antipattern to me, and especially so at insert-time